src/components/custom-field-input-date/custom-field-input-date.jsx

```diff
@@ -78,41 +88,84 @@ const CustomFieldInputDate = forwardRef(function CustomFieldInputDate(props, ref
     return props.errorText;
   };
 
+  const wrapperRef = useRef(null);
+  const handleDropdownClose = () => {
+    setExpanded(false);
+  };
+  useDropdownClose(wrapperRef, expanded, handleDropdownClose);
+
+  const openCalendar = () => {
+    setExpanded(!expanded);
+  };
+
+  function onInputClick() {
+    setExpanded(!expanded);
+  }
+
+  function onInputKeyDown(event) {
+    if (event.key === 'Enter' || event.key === 'Space') {
+      setExpanded(!expanded);
+    }
+  }
+
+  function onKeyDown(event) {
```

Comment:
Should we also capture `keyDown` events from the `Calendar`, to support this same effect when the `Calendar` is focused? 🤔

Reply:
We do handle this when the calendar is focused; that is accomplished with `handleDropdownClose`.